exp/api/v1beta1/azuremanagedmachinepool_webhook_test.go

```diff
@@ -254,6 +254,87 @@ func TestAzureManagedMachinePoolUpdatingWebhook(t *testing.T) {
 			},
 			wantErr: true,
 		},
+		{
+			name: "custom headers are immutable",
+			old: &AzureManagedMachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"infrastructure.cluster.x-k8s.io/custom-header-SomeFeature": "true",
+					},
+				},
+				Spec: AzureManagedMachinePoolSpec{},
+			},
+			new: &AzureManagedMachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"infrastructure.cluster.x-k8s.io/custom-header-SomeFeature": "false",
+					},
+				},
+				Spec: AzureManagedMachinePoolSpec{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "custom headers are immutable 2",
+			old: &AzureManagedMachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						"infrastructure.cluster.x-k8s.io/custom-header-SomeFeature": "true",
+					},
+				},
+				Spec: AzureManagedMachinePoolSpec{},
+			},
+			new: &AzureManagedMachinePool{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{},
+				},
+				Spec: AzureManagedMachinePoolSpec{},
+			},
+			wantErr: true,
+		},
+		{
+			name: "custom headers are immutable 3",
```

Comment:
nit: can we be more specific with the naming?

Reply:
fixed